cmake/OpenCVDownload.cmake

```diff
@@ -1,30 +1,42 @@
 #
 #  Download and optionally unpack a file
 #
-#  ocv_download(FILENAME p HASH h URL u1 [u2 ...] DESTINATION_DIR d [STATUS s] [UNPACK] [RELATIVE])
+#  ocv_download(FILENAME p HASH h URL u1 [u2 ...] DESTINATION_DIR d [ID id] [STATUS s] [UNPACK] [RELATIVE_URL])
 #    FILENAME - filename
 #    HASH - MD5 hash
 #    URL - full download url (first nonempty value will be chosen)
 #    DESTINATION_DIR - file will be copied to this directory
+#    ID     - identifier for project/group of downloaded files
 #    STATUS - passed variable will be updated in parent scope,
 #             function will not fail the build in case of download problem if this option is provided,
 #             but will fail in case when other operations (copy, remove, etc.) failed
 #    UNPACK - downloaded file will be unpacked to DESTINATION_DIR
 #    RELATIVE_URL - if set, then URL is treated as a base, and FILENAME will be appended to it
 #  Note: uses OPENCV_DOWNLOAD_PATH folder as cache, default is <opencv>/.cache
 
-set(OPENCV_DOWNLOAD_PATH "${OpenCV_SOURCE_DIR}/.cache" CACHE PATH "Cache directory for downloaded files")
+set(HELP_OPENCV_DOWNLOAD_PATH "Cache directory for downloaded files")
+if(DEFINED ENV{OPENCV_DOWNLOAD_PATH})
+  set(OPENCV_DOWNLOAD_PATH "$ENV{OPENCV_DOWNLOAD_PATH}" CACHE PATH "${HELP_OPENCV_DOWNLOAD_PATH}")
```

Comment:
Shouldn't we FORCE setting the variable in this case? Otherwise it will have lower priority than the default value on consequent builds.